lib/model/app_model/decode_invoice.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:cln_common/cln_common.dart';
+import 'package:clnapp/model/app_model/app_utils.dart';
+
+class AppDecodeInvoice {
+  DecodeInvoice invoice;
+  AppDecodeInvoice(this.invoice);
+
+  factory AppDecodeInvoice.fromJSON(Map<String, dynamic> json,
+      {bool snackCase = false}) {
+    LogManager.getInstance.debug("RESPONSE HERE : $json");
+    var decodedInvoice = DecodeInvoice.fromJSON(json);
+    return AppDecodeInvoice(decodedInvoice);
+  }
+}
+
+class DecodeInvoice {
+  String description;
+
+  String expirationTime;
+
+  String pubKey;
+
+  String amount;
+
+  DecodeInvoice(
+      {required this.description,
+      required this.amount,
+      required this.expirationTime,
+      required this.pubKey});
+
+  factory DecodeInvoice.fromJSON(Map<String, dynamic> json,
+      {bool snackCase = false}) {
+    var description = json.withKey("description", snackCase: true);
```

Comment:
```suggestion
    var description = json.withKey("description", snackCase:  snackCase);
```

all the item need to have the snackCase otherwise the code will crash when the snackCase == false